themes/ohcoe/static/js/scripts.js

```diff
@@ -29,51 +39,70 @@ $(function() {
-            $(resultsContainer).append(tableRow);
-            domainScore += learningObjectiveScore;
-            learningObjectiveCounter += 1;
+    /* Display Results to Users */
+    var resultsContainer = $('.domain-results tbody');
+    var cumulativePreScore = new Array();
+    var cumulativePostScore = new Array();
+    $('.domain-results').each(function(idx, domainResults){
+        var currentDomain = null;
+        if (domainResults) {
+            currentDomain = domainResults.id;
+        }
+        var domainScore = 0;
+        var learningObjectiveCounter = 0;
+
+        [...Array(3).keys()].map((el) => {
+            var q = el + 1;
+            var prefix = currentDomain + '-' + q;
+            var preScore = localStorage.getItem(prefix + '-pre-q1');
+            var postScore = localStorage.getItem(prefix + '-post-q1');
+            var learningObjectiveScore = Number(postScore) - Number(preScore);
+            if (preScore && postScore) {
+                var tableRow = $('<tr></tr>');
+                tableRow.html('<td>' + localStorage.getItem(prefix) + '</td>' +
+                              '<td>' + preScore + '</td>' +
+                              '<td>' + postScore + '</td>' +
+                              '<td>' + learningObjectiveScore + '</td>');
+                resultsContainer[idx].append(tableRow[0]);
+                domainScore += learningObjectiveScore;
+                learningObjectiveCounter += 1;
+                cumulativePreScore.push(Number(preScore));
+                cumulativePostScore.push(Number(postScore));
+            }
+        });
+
+        // Calulate the percentage of growth for the entire domain
+        var domainScoreGrowth = 0;
+        if (learningObjectiveCounter) {
+            domainScoreGrowth = round(domainScore / learningObjectiveCounter);
+        }
+
+        var scoreContainer = $('#domain-score-' + currentDomain);
+        if (scoreContainer && domainScoreGrowth) {
+            scoreContainer.html(
+                'You have this much growth: <span id="score-container">' +
+                domainScoreGrowth + ' out of 4 </span>');
         }
     });
 
-    // Calulate the percentage of growth for the entire domain
-    var domainScoreGrowth = 0;
-    if (learningObjectiveCounter) {
-        domainScoreGrowth = Math.round(
-            (domainScore / (learningObjectiveCounter * 4)) * 100
-        );
-    }
+    // Progress Bar
+    var meanPreScore = round(mean(cumulativePreScore));
+    var meanPostScore = round(mean(cumulativePostScore));
+    var meanPrePct = (meanPreScore / 4) * 100;
+    var meanPostPct = (meanPostScore / 4) * 100;
 
-    var scoreContainer = $('#domain-score-' + currentDomain);
-    if (scoreContainer && domainScoreGrowth) {
-        scoreContainer.html(
-            'You have this much growth: <span id="score-container">' +
-            domainScoreGrowth + '%</span>');
-    }
+    var preScorePBar = $('#cumulative-pre-score')[0];
```

Comment:
The element referenced as `#cumulative-pre-score` is undefined.

Reply:
Got it, I fixed it by calling the methods to render this new feature only on the page where its needed.